app/server/lib/Archive.ts

```diff
@@ -11,11 +11,27 @@ export interface ArchiveEntry {
   data: stream.Readable | Buffer;
 }
 
+export interface ArchivePackingOptions {
+  // Whether the destination stream should be closed once the archive has been written.
+  endDestStream: boolean;
+}
+
+const defaultPackingOptions: ArchivePackingOptions = {
+  endDestStream: true,
+};
+
 export interface Archive {
   mimeType: string;
   fileExtension: string;
-  dataStream: stream.Readable;
-  completed: Promise<void>;
+  /**
+   * Starts packing files into the archive.
+   * This will block indefinitely if dataStream is never read from.
+   * This resolves when all files are processed, or an error occurs.
+   * This will not close the destination stream. This is to allow the caller to decide how to handle
```

Comment:
It _does_ close the stream by default, right?

Reply:
Good catch - wrote the comment before I parameterised this!

Reply:
Removed these lines - it's documented on the options interface above!